services/capture/src/main/java/ca/bc/gov/tno/services/capture/handlers/ScheduledService.java

```diff
@@ -0,0 +1,155 @@
+package ca.bc.gov.tno.services.capture.handlers;
+
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.Date;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Component;
+import org.springframework.context.ApplicationEvent;
+
+import ca.bc.gov.tno.dal.db.services.interfaces.IDataSourceService;
+import ca.bc.gov.tno.dal.db.services.interfaces.IMediaTypeService;
+import ca.bc.gov.tno.services.ServiceState;
+import ca.bc.gov.tno.services.data.BaseDbScheduleService;
+import ca.bc.gov.tno.services.data.config.DataSourceCollectionConfig;
+import ca.bc.gov.tno.services.data.config.DataSourceConfig;
+import ca.bc.gov.tno.services.data.config.ScheduleConfig;
+import ca.bc.gov.tno.services.data.events.TransactionBeginEvent;
+import ca.bc.gov.tno.services.capture.config.CaptureConfig;
+import ca.bc.gov.tno.services.capture.config.CaptureCollectionConfig;
+
+/**
+ * ScheduledService class, provides a process that manages the scheduling of
+ * ingestion. This will ensure the process runs according to the configured
+ * schedule.
+ */
+@Async
+@Component
+public class ScheduledService
+    extends BaseDbScheduleService<CaptureConfig, DataSourceCollectionConfig<CaptureConfig>> {
+
+  private static final Logger logger = LogManager.getLogger(ScheduledService.class);
+  private final CaptureConfig mediaConfig;
+  private final IMediaTypeService mediaTypeService;
+
+  /**
+   * Creates a new instance of a ScheduledService object, initializes with
+   * specified parameters.
+   * 
+   * @param state             Service state.
+   * @param mediaConfig       Capture media type config.
+   * @param config            Capture config.
+   * @param dataSourceService DAL DB data source service.
+   * @param mediaTypeService  DAL DB media type service.
+   * @param eventPublisher    Application event publisher.
+   */
+  @Autowired
+  public ScheduledService(final ServiceState state,
+      final CaptureConfig mediaConfig,
+      final CaptureCollectionConfig config,
+      final IDataSourceService dataSourceService,
+      final IMediaTypeService mediaTypeService,
+      final ApplicationEventPublisher eventPublisher) {
+    super(state, config, dataSourceService, eventPublisher);
+    this.mediaConfig = mediaConfig;
+    this.mediaTypeService = mediaTypeService;
+  }
+
+  /**
+   * Update the data source with the current ranAt date and time.
+   * 
+   * @param config The data source config.
+   * @param ranAt  The date and time the transaction ran at.
+   */
+  @Override
+  protected void updateDataSource(DataSourceConfig dataSource, ScheduleConfig schedule, ZonedDateTime ranOn) {
```

Comment:
At some point in the future we should clean this logic up so that we don't have to save twice.

Reply:
This is the issue I was talking about yesterday. Do I save once in ScheduledService and override the functionality of the superclass (by not calling it) or do we save twice?

Reply:
yes, at some point we can redesign.  Technically it's not an issue to save twice presently, it's just a bad implementation.